src/factom-event-emitter.js

```diff
@@ -0,0 +1,274 @@
+    //                      EVENT EMITTERS                      //
+    /////////////////////////////////////////////////////////////
+
+    // Emit factoid block and/or trigger factoid transaction emitter function
+    async _emitFactoidBlock(directoryBlock) {
+        try {
+            const factoidBlock = await this._cli.getFactoidBlock(directoryBlock.factoidBlockRef);
+
+            if (this.listenerCount(FACTOM_EVENT.factoidBlock)) {
+                this.emit(FACTOM_EVENT.factoidBlock, factoidBlock);
+            }
+
+            if (this._factoidAddressSubscriptions.size) {
+                // Must pass directoryBlock through to access block context
+                this._emitFactoidTransaction(factoidBlock, directoryBlock);
+            }
+        } catch (err) {
+            this.emit('error', err);
+        }
+    }
+
+    async _emitAdminBlock(directoryBlock) {
+        try {
+            const adminBlock = await this._cli.getAdminBlock(directoryBlock.adminBlockRef);
+            this.emit(FACTOM_EVENT.adminBlock, adminBlock);
+        } catch (err) {
+            this.emit('error', err);
+        }
+    }
+
+    async _emitEntryCreditBlock(directoryBlock) {
+        try {
+            const entryCreditBlock = await this._cli.getEntryCreditBlock(directoryBlock.entryCreditBlockRef);
+            this.emit(FACTOM_EVENT.entryCreditBlock, entryCreditBlock);
+        } catch (err) {
+            this.emit('error', err);
+        }
+    }
+
+    // Emit the first entry block of any newly created entry chain.
+    async _emitNewEntryChains(directoryBlock) {
+        try {
+            const checkIfNewChainAndEmit = async ref => {
+                const entryBlock = await this._cli.getEntryBlock(ref.keyMR);
+                if (entryBlock.sequenceNumber === 0) {
+                    this.emit(FACTOM_EVENT.newEntryChain, entryBlock);
+                }
+            };
+
+            // Fetch entry chains contained in directory block concurrently
+            await Promise.map(directoryBlock.entryBlockRefs, checkIfNewChainAndEmit, { concurrency: 5 });
+        } catch (err) {
+            this.emit('error', err);
+        }
+    }
+
+    // Emit the latest entry block of any entry chain the user is listening to.
+    async _emitEntryBlock(directoryBlock) {
+        const fetchAndEmitNewBlock = async chainId => {
+            try {
+                const entryBlockRef = directoryBlock.entryBlockRefs.find(ref => ref.chainId === chainId);
+
+                if (entryBlockRef) {
+                    const entryBlock = await this._cli.getEntryBlock(entryBlockRef.keyMR);
+                    this.emit(chainId, entryBlock);
+                }
+            } catch (err) {
+                this.emit('error', err);
+            }
+        };
+
+        // Fetch new entry blocks concurrently
+        await Promise.map(this._entryChainSubscriptions, fetchAndEmitNewBlock, { concurrency: 5 });
```

Comment:
Is there any particular reason to choose 5? I feel a factomd instance could easily take more concurrent request, from my experience/intuition. Maybe raise it to 10 at leat?